includes/models/newspaper.inc

```diff
@@ -1,42 +1,99 @@
 <?php
-
-
 /**
  * @file
  */
-
 /**
- * This is the actual manifest building function.
+ * Create the requested newspaper manifest. When $selected_collection is set, the manifest of that collection is returned otherwise
+ * the full news paper manifest collection is returned
+ * @param AbstractObject $object
+ * @param null $selected_collection - Return the selected sub collection
+ * @return mixed
  */
-function islandora_iiif_manifest_create_newspaper_manifest(AbstractObject $object) {
+function islandora_iiif_manifest_create_newspaper_manifest(AbstractObject $object, $selected_collection = NULL) {
+  
   module_load_include('inc', 'islandora', 'includes/utilities');
   module_load_include('inc', 'islandora_iiif_manifests', 'includes/utilities');
   module_load_include('inc', 'islandora_paged_content', 'includes/utilities');
   module_load_include('inc', 'islandora_newspaper', 'includes/utilities');
+  module_load_include('inc', 'islandora', 'includes/datastream');
+  
+  if (is_null($selected_collection)) {
+    $manifest = islandora_iiif_manifest_get_newspaper_manifest($object);
+  }
+  else {
+    $manifest = islandora_iiif_manifest_get_newspaper_collection($object, $selected_collection);
+  }
+  
+  return $manifest;
+}
 
-  // Manifest array.
-  $manifest = array();
-
+/**
+ * Create the newspaper manifest with collections and its member within
+ * @param $object
+ * @return mixed
+ */
+function islandora_iiif_manifest_get_newspaper_manifest($object) {
+  // Collections array
+  $manifest['collections'] = array();
   // Get issues.
   $issues = islandora_newspaper_get_issues($object);
-
-  $collection_members_members = array();
-
-  foreach ($issues as $issue) {
-
-    // Load issue.
-    $issue_object = islandora_object_load($issue['pid']);
-
-    $collection_item = islandora_iiif_manifests_get_object_collection($issue_object);
-
-    $collection_members_members[] = $collection_item['object'];
-
+  // Get issues.
+  $group_issues = islandora_newspaper_group_issues($issues);
```

Comment:
This function returns an associated array: year is key, value is array with keys are months, value is array with keys are days, value is issue? Make this more clear.